### PR TITLE
Watcher accounts constructed lazily

### PR DESCRIPTION
**Pull request by albertzaharovits**
This fixes two bugs about watcher notifications:

* registering accounts that had only secure settings was not possible before; these accounts are very much practical for `Slack` and `PagerDuty` integrations.

* removes the limitation that, for an account with both secure and cluster settings, the admin had to first change/add the secure settings and only then the dependent dynamic cluster settings. The reverse order would trigger a `SettingsException` for an incomplete account. This had been partially discussed in: https://github.com/elastic/elasticsearch/pull/36403#discussion_r240036479

The workaround is to lazily instantiate account objects. Previously the approach was to greedily validate all the account settings by constructing the account objects even if they would not ever be used by actions. This made sense in a world where all the settings were set by a single API. But given that accounts have dependent settings (that must be used together) that have to be changed using different APIs (`POST _nodes/reload_secure_settings` and `PUT _cluster/settings`), the settings could be in an invalid state in between the calls.
This fix only builds account objects (and validates the settings) when they are needed by actions.

### CODE REVIEW

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/NotificationServiceTests.java [hunk 1]
```diff
@@ -80,16 +95,135 @@ public void testAccountDoesNotExist() throws Exception{
...
+        settingValue.set(randomAlphaOfLength(4));
+        settingsBuilder.put(setting.getKey(), settingValue.get());
+        service.clusterSettingsConsumer(settingsBuilder.build());
+        assertThat(validationInvocationCount.get(), is(2));
+        service.getAccount(null);
+        assertThat(validationInvocationCount.get(), is(3));
+        // update both
+        if (randomBoolean()) {
+            // update secure first
+            secureSettingValue.set(randomAlphaOfLength(4));
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/NotificationServiceTests.java]: putting these three line calls in dedicated methods allows you to get rid of the "first/second" comments and its easy to figure out the order when reading

**Comment by spinscale** [comment]
does this PR also justify a backport into 6.5?

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/NotificationServiceTests.java (hunk 1 above)
Comment (albertzaharovits) [reply x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/NotificationServiceTests.java]: I did the replacement but parameters have side effects and the functions look super ugly.

**Comment by albertzaharovits** [comment]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/2425/console
```
19:50:22 > Task :x-pack:plugin:watcher:test FAILED
19:50:22 Tests with failures:
19:50:22   - org.elasticsearch.xpack.watcher.notification.hipchat.HipChatServiceTests.testSingleAccountIntegration
19:50:22   - org.elasticsearch.xpack.watcher.notification.hipchat.HipChatServiceTests.testMultipleAccounts
19:50:22   - org.elasticsearch.xpack.watcher.notification.hipchat.HipChatServiceTests.testSingleAccountV1
19:50:22   - org.elasticsearch.xpack.watcher.notification.hipchat.HipChatServiceTests.testSingleAccountIntegrationNoRoomSetting
19:50:22   - org.elasticsearch.xpack.watcher.notification.hipchat.HipChatServiceTests.testSingleAccountUser
```
Was a legit failure!

The problem is that Accounts were not cached, every request for an account rebuilded it. I have swapped `Supplier` for `LazyInitializable` to remediate, and added a test. I think it's good now!

**Review by spinscale — APPROVED** [review]
LGTM. Thanks for picking this up!

**Pull request merged**